### ISSUE TITLE
automatic displaying tv channels in the service list with lcn 1000, 1001 and...

### ISSUE THREAD

**Issue by ffallahi6303** [DVBproject/DVB-I-Reference-Client#21]
now, the client shows tv channels in the service list with lcn 1000, 1001,... it makes the viewer confused.
Is it possible to have the list of these channels in another key? for example blue. and tv would show this list by pressing the blue button.

**Comment by sofia-tsa** [comment]
We could add an option to the settings to show or hide services that are not listed in the DVB-I service list.

**Comment by paulhiggs** [comment]
I may have mentioned in another issue that it could be useful to have a config file (i.e. `config.js`) that contains some app configuration items.
This one could be

`const INCLUDE_LOCAL_BROADCAST_CHANNELS=true;`

**Comment by jpiesing** [comment]
Why not include these in the DVB-I service list with an appropriate LCN?

**Comment by paulhiggs** [comment]
> Why not include these in the DVB-I service list with an appropriate LCN?

That would be the "expected behaviour", however in labs and other test situations, the app provider my want to include only a subset (0 to all) of the local scanned broadcast channels and hide the rest. A flag as I propose could support this, otherwise just comment out the line related code from `frontend/hbbtv/launcher/index.php` (look for `number = 1000;`)

**Comment by paulhiggs** [comment]
related to issue 14

**Comment by ffallahi6303** [comment]
> Why not include these in the DVB-I service list with an appropriate LCN?

I am thinking what is the good service list.  our viewer used to have TV1 in lcn 1. then what should I do for a service list if it should be used as a common list for dvb-t and dash services? i have to set lcn 1 for dvb-t service(TV1). then I have my dvb-t services in service list. why do I repeat it again in the list with lcn 1000? 

or another option is to have just ott services in the service list and assign suitable lcn ( those are not used in dvb-t) to them. then no common dvb-t and ott list.

**Comment by ffallahi6303** [comment]
> > Why not include these in the DVB-I service list with an appropriate LCN?
> 
> That would be the "expected behaviour", however in labs and other test situations, the app provider my want to include only a subset (0 to all) of the local scanned broadcast channels and hide the rest. A flag as I propose could support this, otherwise just comment out the line related code from `frontend/hbbtv/launcher/index.php` (look for `number = 1000;`)



actually, in lab environment, i can scan just my channel and i don't have this problem, but I have problem with a real environment and implementation.

**Comment by paulhiggs** [comment]
> > Why not include these in the DVB-I service list with an appropriate LCN?
> 
> I am thinking what is the good service list. our viewer used to have TV1 in lcn 1. then what should I do for a service list if it should be used as a common list for dvb-t and dash services? 

You can certainly create a service list with LCN=1 for TV1 which would have two <ServiceInstance> elements (one for DVB-T and one for DVB-DASH). On a television running the HbbTV app, the DVB-T instance would be selected and on a mobile phone running the Android app, the DVB-DASH instance would be selected. Both people see TV1 on LCN 1.

> i have to set lcn 1 for dvb-t service(TV1). then I have my dvb-t services in service list. why do I repeat it again in the list with lcn 1000?

You do not need to dd those DVB-T services again at LCN1000+. Channel numbers 1000 and above are only used for channels reported by the television that **are not** represented in the service list - that match is based on Triplet and should be according to clause 5.2.1 of [A177](http://dvb.org/wp-content/uploads/2019/11/A177_DVB-I_Nov_2019.pdf)
> 
> or another option is to have just ott services in the service list and assign suitable lcn ( those are not used in dvb-t) to them. then no common dvb-t and ott list.

This is not necessary unless you want different DVB-DASH services for mobile and TV devices

**Comment by ffallahi6303** [comment]
You do not need to dd those DVB-T services again at LCN1000+. Channel numbers 1000 and above are only used for channels reported by the television that are not represented in the service list - that match is based on Triplet and should be according to clause 5.2.1 of A177



Right now i checked it again. the app shows service even if it present in the service list by Triplet.

**Comment by sofia-tsa** [comment]
Does the service have multiple service instances, DASH and DVB? Looks like there is a bug, if the dvb instance isn't the highest priority service instance then the dvb-channel will be listed in the 1000+  space also. I'll fix that right away.

**Comment by ffallahi6303** [comment]
i have a service with both instances and dvb has higher priority . 

and also another service with only one instance .

both of these services are listing again with lcn +1000.

i guess in example.xml sid1 and sid2 should be demo1 and demo2 . yes?
but their onid and tsid didn't ser correctly. it should be 8438 as onid and 1097 as tsid.
then you can see them again in service listt with +1000.

**Comment by paulhiggs** [comment]
Just a note on the use of triplet for correlating a scanned broadcast service with services defined in the DVB-I Service List.
When this was discussed, was determined that the same scanned broadcast service could be in more than one <Service> element, ether for regional definition or for other purposes, i.e.
````
<LCNTableList>
  <LCNTable>
    <LCN channelNumber="23" serviceRef="tag:dvb.org,2020:tv1"/>
    <LCN channelNumber="33" serviceRef="tag:dvb.org,2020:tv1bis"/>
  </LCNTable>
</LCNTableList>
<Service>
  <UniqueIdentifier>tag:dvb.org,2020:tv1</UniqueIdentifier>
  <ServiceInstance>
    <DVBTDeliveryParameters>
      <DVBTriplet origNetId="318" tsId="5200" serviceId="3403"/>
    </DVBTDeliveryParameters>
  </ServiceInstance>
  <ServiceName>TV 1</ServiceName>
</Service>
<Service>
  <UniqueIdentifier>tag:dvb.org,2020:tv1bis</UniqueIdentifier>
  <ServiceInstance>
    <DVBTDeliveryParameters>
      <DVBTriplet origNetId="318" tsId="5200" serviceId="3403"/>
    </DVBTDeliveryParameters>
  </ServiceInstance>
  <ServiceName>Also TV 1</ServiceName>
</Service>
````
is perfectly valid

**Comment by sofia-tsa** [comment]
I found another issue with the unlisted dvb-channels, causing the services to be listed twice. It is caused by the fix for issue #18 , will be fixed in the next release

**Comment by ffallahi6303** [comment]
I have a demonstration of services on next saturday. Can you please update the code that fix this issue? Then I can work on this new code for my demo.

**Comment by juhajoki** [comment]
we will make a new release before your demo.

**Comment by sofia-tsa** [comment]
Should be fixed in release 2.8. Please verify

**Comment by paulhiggs** [comment]
This issue covered two aspects, (a) the selection of the highest priority service instance and (b) the "hiding" of LCN 1000+ channels (scanned channels not matching DVB-I Services). Are both these items solved by the version 2.8 update?

**Comment by ffallahi6303** [comment]
Selecting highest priority service instance was solved in issue #18 ,(https://github.com/DVBproject/DVB-I-Reference-Client/commit/b8c23ef02578038b394e1b3d3cf917dfc0e548e8)

 I've tested it before. thank you

**Comment by ffallahi6303** [comment]
> Should be fixed in release 2.8. Please verify

it has been fixed in the new version. thank you
